src/models/deployment-service.test.ts

```diff
@@ -0,0 +1,118 @@
+		it("should successfully save deployment info to dynamo db", async () => {
+
+			const createdAt = new Date();
+
+			await saveDeploymentInfo({
+				gitHubBaseUrl: "https://github.com",
+				gitHubInstallationId: 1,
+				repositoryId: 3,
+				commitSha: "abc-abc-abc",
+				description: "some-random description",
+				env: "production",
+				status: "success",
+				createdAt
+			}, logger);
+
+			const result = await ddb.getItem({
+				TableName: envVars.DYNAMO_DEPLOYMENT_HISTORY_TABLE_NAME,
+				Key: {
+					"Id": { "S": `ghurl_https://github.com_ghid_1_repo_3_env_production` },
+					"StatusCreatedAt": { "N": String(createdAt.getTime()) }
+				},
+				AttributesToGet: [
+					"Id", "StatusCreatedAt",
+					"GitHubInstallationId", "GitHubAppId", "RepositoryId",
+					"CommitSha", "Description",
+					"Env", "Status", "ExpiredAfter"
+				]
+			}).promise();
+
+			expect(result.$response.error).toBeNull();
+			expect(result.Item).toEqual({
+				Id: { "S": "ghurl_https://github.com_ghid_1_repo_3_env_production" },
+				StatusCreatedAt: { "N": String(createdAt.getTime()) },
+				GitHubInstallationId: { "N": "1" },
+				RepositoryId: { "N": "3" },
+				CommitSha: { "S": "abc-abc-abc" },
+				Description: { "S": "some-random description" },
+				Env: { "S": "production" },
+				Status: { "S": "success" },
+				ExpiredAfter: { "N": String(Math.floor((createdAt.getTime() + ONE_YEAR_IN_MILLISECONDS) / 1000)) }
+			});
+		});
+	});
+
+	describe("fetching back last success deployment", () => {
+		const gitHubInstallationId = 111;
+		const repositoryId = 222;
+		const createdAt1 = new Date("2000-01-01");
+		const createdAt2 = new Date("2000-02-02");
+		const createdAt3 = new Date("2000-03-03");
+		beforeEach(async () => {
+			await saveDeploymentInfo({
+				gitHubBaseUrl: "https://github.com",
+				gitHubInstallationId, repositoryId,
+				commitSha: "create-1", description: "",
+				env: "production", status: "success",
+				createdAt: createdAt1
+			}, logger);
+			await saveDeploymentInfo({
+				gitHubBaseUrl: "https://github.com",
+				gitHubInstallationId, repositoryId,
+				commitSha: "create-2", description: "",
+				env: "production", status: "success",
+				createdAt: createdAt2
+			}, logger);
+			await saveDeploymentInfo({
+				gitHubBaseUrl: "https://github.com",
+				gitHubInstallationId, repositoryId,
+				commitSha: "create-3", description: "",
+				env: "production", status: "success",
+				createdAt: createdAt3
+			}, logger);
+		});
+		it("should fetch last success deployment", async () => {
+			const result = await findLastSuccessDeployment({
+				gitHubBaseUrl: "https://github.com",
+				gitHubInstallationId, repositoryId,
+				env: "production", currentDate: createdAt2
+			}, logger);
+			expect(result).toEqual({
+				repositoryId,
+				commitSha: "create-1",
+				createdAt: createdAt1
+			});
+		});
+		it("should fetch THE last success deployment when more than one previous success deployments", async () => {
```

Comment:
Could we please also add a test for `findLastSuccessDeployment` when currentDate is not exactly equals the stored value?

Reply:
added.